grug/testing/src/suite.rs

```diff
@@ -645,28 +645,19 @@ where
     AppError: From<DB::Error> + From<VM::Error> + From<PP::Error> + From<ID::Error>,
     Self: Querier,
 {
-    pub fn query_balance<D>(
-        &self,
-        address: &dyn Addressable,
-        denom: D,
-    ) -> Result<Uint128, <Self as Querier>::Error>
+    pub fn query_balance<D>(&self, address: &dyn Addressable, denom: D) -> StdResult<Uint128>
     where
         D: TryInto<Denom>,
-        <D as TryInto<Denom>>::Error: Debug,
+        StdError: From<D::Error>,
     {
-        let denom = denom.try_into().unwrap();
-        self.query_chain(Query::balance(address.address(), denom))
-            .map(|res| res.as_balance().amount)
+        let address = address.address();
+        let denom = denom.try_into()?;
+        <Self as QuerierExt>::query_balance(self, address, denom)
     }
 
-    pub fn query_balances(&self, account: &dyn Addressable) -> AppResult<Coins> {
-        self.app
-            .do_query_app(
-                Query::balances(account.address(), None, Some(u32::MAX)),
-                0, // zero means to use the latest height
-                false,
-            )
-            .map(|res| res.as_balances())
+    pub fn query_balances(&self, account: &dyn Addressable) -> StdResult<Coins> {
+        let address = account.address();
+        <Self as QuerierExt>::query_balances(self, address, None, Some(u32::MAX))
```

Comment:
@Rhaki These two functions are bad:

1. They have the same names as the ones in `QuerierExt`, which is confusing.
2. Inconsistency:
   - in one function, the input is named `address`, while in the other it's named `account`
   - one uses `query_chain`, the other uses `do_query_app`
3. You used `unwrap`

I'd rather delete them, but some tests still use them, so let's do this later.